Solved issues with AWS CentOS 7.

jdelaporte opened the pull request:
Hi Greg,

Thanks for publishing this role on GitHub! 

I had some issues running it on an AWS instance with CentOS 7 installed. On my machine, hostname module didn't work. Also, firewalld was not installed by default. I made a couple changes that worked for me, so I am sending them back to you in case you want to incorporate them

Thanks,
Joanna

gregswift commented on tasks/firewalld.yml~:
```diff
@@ -0,0 +1,28 @@
+---
```
Can you remove this file from your commit?

gregswift commented on tasks/main.yml~:
```diff
@@ -0,0 +1,49 @@
+---
+#- name: Hard set the hostname
```
can you remove this file from your PR

gregswift commented on tasks/main.yml:
```diff
@@ -2,6 +2,8 @@
 - name: Hard set the hostname
   hostname: name={{ ansible_fqdn }}
 
+# - shell: hostname {{ ansible_fqdn }}
```
Can you remove this?

jdelaporte replied:
Sure. Give me a few minutes while I figure out how. I am painfully new to
git pull requests still.

On Mon, Nov 23, 2015 at 4:33 PM, Greg Swift <notifications@github.com>
wrote:

> In tasks/main.yml
> <https://github.com/gregswift/ansible-ipaserver/pull/1#discussion_r45671176>
> :
>
> > @@ -2,6 +2,8 @@
> >  - name: Hard set the hostname
> >    hostname: name={{ ansible_fqdn }}
> >
> > +# - shell: hostname {{ ansible_fqdn }}
>
> Can you remove this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gregswift/ansible-ipaserver/pull/1/files#r45671176>.
>



-- 


Joanna Delaporte
Linux Systems Administrator | Parkland College
joannadelaporte@gmail.com

jdelaporte commented:
I think I cleaned it up successfully. Let me know if it doesn't look right.

gregswift commented:
Joanna,

I'm down with brining your firewall changes in.  I think i'm gonna turn around and add some *when* logic to them, but they are a great addition.  If you could clean the other bits out that would be awesome.

thanks

jdelaporte commented:
Hi Greg,

What other bits? Do you mean the extra backup files? Or some comments or
something in the firewall.yml file? I'm offline now, so I can't see it at
the moment.

Thanks!
Joanna
On Nov 23, 2015 4:53 PM, "Greg Swift" <notifications@github.com> wrote:

> Joanna,
>
> I'm down with brining your firewall changes in. I think i'm gonna turn
> around and add some *when* logic to them, but they are a great addition.
> If you could clean the other bits out that would be awesome.
>
> thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gregswift/ansible-ipaserver/pull/1#issuecomment-159093430>
> .
>

gregswift commented on tasks/firewalld.yml:
```diff
@@ -1,10 +1,20 @@
   when: ansible_distribution == "Fedora"
 
-- name: Enable the individual rulesets for CentOS
+- name: Enable the individual rulesets for CentOS 
```
some trailing whitespace made it in

gregswift commented:
The 'other bits' was just the other commetns i made. it all looks great

jdelaporte commented:
Hi Greg, 

I added a basic RedHat family conditional to the yum install firewalld. You might want to make it more specific. I cleaned out a couple whitespaces I found, too.

gregswift commented on tasks/firewalld.yml:
```diff
@@ -1,4 +1,14 @@
+#firewalld doesn't exist on AWS CentOS 7 by default
+- name: Make sure firewalld is installed
+  yum: name=firewalld state=present
+  when: ansible_os_family == "RedHat"
```
I dont think we need to add the when here. other families have firewalld too, but it was a good thought

jdelaporte replied:
Right, but they don't all have yum.
On Nov 23, 2015 11:01 PM, "Greg Swift" <notifications@github.com> wrote:

> In tasks/firewalld.yml
> <https://github.com/gregswift/ansible-ipaserver/pull/1#discussion_r45697191>
> :
>
> > @@ -1,4 +1,14 @@
> >  ---
> > +#firewalld doesn't exist on AWS CentOS 7 by default
> > +- name: Make sure firewalld is installed
> > +  yum: name=firewalld state=present
> > +  when: ansible_os_family == "RedHat"
>
> I dont think we need to add the when here. other families have firewalld
> too, but it was a good thought
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gregswift/ansible-ipaserver/pull/1/files#r45697191>.
>

gregswift replied:
Touche. Good point.

gregswift commented:
I've updated the role with a wrapper boolean on the firewalld subtasks.  However I need to test this before i cut a release. not sure when i'll get to that.
